File: liqp-core/src/main/java/liqp/filters/Ceil.java
package liqp.filters;

import liqp.nodes.RenderContext;

public class Ceil extends Filter {

    @Override
    public Object apply(RenderContext context, Object value, Object... params) {

        if (!super.isNumber(value)) {
            return value;
        }

        return (long)Math.ceil(super.asNumber(value).doubleValue());
    }
}

File: liqp-core/src/main/kotlin/liqp/filter/Ceil.kt
package liqp.filter

import liqp.context.LContext

/**
 * {{ 4 | ceil }} -> 4
 * {{ 4.3 | ceil }} -> 5
 */
class Ceil : LFilter() {
  override fun onFilterAction(context: LContext, value: Any?, params: FilterParams): Any? {
    context.run {
      return Math.ceil(asDouble(value) ?: 0.0).toLong()
    }
  }
}
